docs/vfio.md

```diff
@@ -0,0 +1,81 @@
+# Cloud Hypervisor VFIO HOWTO
+
+VFIO (Virtual Function I/O) is a kernel framework that exposes direct device
+access to userspace. `cloud-hypervisor`, as many VMMs do, uses the VFIO
+framework to directly assign host physical devices to the guest workloads.
+
+## Direct Device Assignment with Cloud Hypervisor
+
+To assign a device to a `cloud-hypervisor` guest, the device needs to be managed
+by the VFIO kernel drivers. However, by default, a host device will be bound to
+its native driver, which is not the VFIO one.
+
+As a consequence, a device must be unbound from its native driver before passing
+it to `cloud-hypervisor` for assigning it to a guess.
+
+### Example
+
+In this example we're going to assign a PCI memory card (SD, MMC, etc) reader
+from the host in a cloud hypervisor guest.
+
+`cloud-hypervisor` only supports assigning PCI devices to its guests. `lspci`
+helps with identifying PCI devices on the host:
+
+```
+$ lspci
+[...]
+01:00.0 Unassigned class [ff00]: Realtek Semiconductor Co., Ltd. RTS525A PCI Express Card Reader (rev 01)
+[...]
+```
+
+Here we see that our device is on bus 1, slot 0 and function 0 (`01:00.0`).
+
+Now that we have identified the device, we must unbind it from its native driver
+(`rtsx_pci`) and bind it to the VFIO driver instead (`vfio_pci`).
+
+First we add VFIO support to the host:
+
+```
```

Comment:
```
$ sudo modprobe vfio_iommu_type1 allow_unsafe_interrupts
```
is needed in the qemu use case for instance. I'm not sure about the reason though. Might be worth adding it, WDYT?

Reply:
Added as well.